R/interact.R

```diff
@@ -106,7 +106,12 @@ run_safely <- function(file) "dummy"
 #' Dummy function for running a file
 #' @noRd
 run_file <- function(file){
-   source(file, local = TRUE)
+   if (is.null(getOption("log.rx.exec.env"))){
+      exec_env <- new.env(parent=globalenv())
+   } else{
+      exec_env <- getOption("log.rx.exec.env")
+   }
+   source(file, local = exec_env)
```

Comment:
Could we just `source` the file w/the default `local = FALSE`?  This will run in the global environment instead of being a child of the global environment.  

```suggestion
   source(file)
```

Reply:
So @mstackhouse can talk to this as well but we wanted to run in a child of global so that it didn't leave anything in the global after the run.  I was going to outline that you can set the option to FALSE if you want it to run in global.

Reply:
@nicholas-masel check out #104 - I try to provide an explanation of what I was looking for in that original issue. 

There's a lot of ways that `logrx::axecute()` could be used. For example, as a driver script for running multiple programs. Maybe in the future it could drive parallel execution of multiple programs at once (which was a thought that I had last night). But using SAS submission tools in the past, one of things that I've _really_ fought with was programs running sequentially but leaving side effects based on the order in which the were executed with whatever SAS process was batching the process. As an example, I once tracked down a SAS program that had VALIDVARNAME = UPPER set, which had a trickle effect 8 programs later when they were batched together.

The thought I was going for here was - if I `axecute()` program a, and then I `axecute()` program b, absolutely nothing that happened in program a should impact program b. So in essence, the program should behave as if the session was completely clean, as if your session was restarted.

This doesn't get us _all the way_ there, because you'd still inherit with global, but it starts to clean up side effects. 

Just a thought, based on what @kodesiba put together here, you could run in global by doing `option(log.rx.exec.env = globalenv())`.

Reply:
Thanks @kodesiba and @mstackhouse ! This makes sense to me now :)  

I ran a few tests and I see the environment we create is cleaned up.  I was confused where so I dug in a bit and see it only exits within [`loudly()`](https://github.com/atorus-research/logrx/blob/3b4931116f80b854df914fce233c73b4fe210e10/R/utils-execution.R#L30) and is gone after.

If we create the environment and not the user, should we just go ahead and clean this up this immediately after `source(file, local = exec_env)` or is this ok as is since it is removed?